### PR TITLE
docs: overhauled documentation

### PR DESCRIPTION
**Pull request by callmehiphop**
### _Big Update Ahead!_

> This PR is one (of several) to move gcloud-node to the new gcloud-common docs app.
#### Notable changes:
- Completely removed old docs site
- Created a manifest.json file for the new site
- Updated all `@example` tags to follow JSDoc standard
- Created a new script to generate documentation JSON
### TODOS
- [x] Remove `@linkcode` tags in favor for custom type syntax (`{module:service/object#method}`)
- [x] Correct improper use of `@mixes` tag for ACL and IAM properties
- [x] Update script to parse example comment syntax
- [x] Move overviews to @classdesc tags
- [x] Re-generate JSON for previous versions (separate PR to gh-pages) (https://github.com/GoogleCloudPlatform/gcloud-node/pull/1098)
- [x] version markdown files
- [x] Wait for new doc app to be merged into gcloud-common (https://github.com/GoogleCloudPlatform/gcloud-common/pull/34)

@stephenplusplus let me know if there's anything I can do to make this easier to go through.. :)

### CODE REVIEW

**Comment by stephenplusplus** [comment]
> Correct improper use of @mixes tag for ACL and IAM properties

What is the improper use?

**Comment by callmehiphop** [comment]
> What is the improper use?

I think `@mixin` implies that we're inheriting either IAM or ACL, but we're actually just storing an instance as a property.. little bit different.

**Comment by stephenplusplus** [comment]
@mixin: A mixin provides functionality that is intended to be added to other objects. If desired, you can use the @mixin tag to indicate that an object is a mixin. You can then add the @mixes tag to objects that use the mixin.

JSDocs is pretty wild. What's a better way to describe (and document) the connection?

**Comment by callmehiphop** [comment]
> What's a better way to describe (and document) the connection?

Still looking into that.. worst case scenario I'll just make `@mixin` work somehow!

**Comment by stephenplusplus** [comment]
Cool cool.

**Comment by callmehiphop** [comment]
But if I can make it work without it, one thing I'd strive to do is kill off all the [empty variables](https://github.com/GoogleCloudPlatform/gcloud-node/blob/master/lib/storage/bucket.js#L283)..

**Comment by stephenplusplus** [comment]
Yes, please!

File: lib/index.js [hunk 1]
```diff
@@ -317,9 +317,6 @@ var scopedApis = {
  * 2. Provide them at the time of instantiation of sub-modules, e.g. a Datastore
  * dataset, a Cloud Storage bucket, etc.
  *
- * See our [Authentication Guide](#/authentication) for how to obtain the
```
Comment (callmehiphop) [inline lib/index.js]: I took this out because all the guides are now tied to a version

File: lib/datastore/transaction.js [hunk 2]
```diff
@@ -353,20 +375,25 @@ Transaction.prototype.delete = function(entities) {
...
- *     data: {
- *       HQ: 'Dallas, TX'
- *     }
- *   },
- *   {
- *     key: productKey,
- *     data: {
- *       vendor: 'Dell'
+ *
+ * dataset.runInTransaction(function(transaction, done) {
```
Comment (callmehiphop) [inline lib/datastore/transaction.js]: An unfortunate side effect to generating the new JSON schema is that we filter out all private examples, so we'll need to test transactions via `dataset.runInTransaction` now :(((

File: docs/manifest.json [hunk 3]
```diff
@@ -0,0 +1,260 @@
+{
+  "lang": "node",
+  "friendlyLang": "Node.js",
+  "markdown": "javascript",
+  "versions": [
```
Comment (callmehiphop) [inline docs/manifest.json]: @stephenplusplus any thoughts on how we should go about programmatically updating this?

Comment (stephenplusplus) [inline docs/manifest.json]: Maybe we should have a release script that:

1. Runs linter
2. Builds docs
3. Runs unit tests
4. Runs system tests
5. Makes release tag via `npm version`
6. Updates `manifest.json` & commits changes
7. Pushes (with tags) to master
8. Runs `npm publish`

--

Or, could the FE just ping GH's API for the versions? Maybe like how it currently gets the latest version for the homepage? `(Latest Release v0.27.0 Dec 22, 2015)`

Comment (callmehiphop) [inline docs/manifest.json]: > Or, could the FE just ping GH's API for the versions? Maybe like how it currently gets the latest version for the homepage?

Oddly enough I tried this and ran into rate limiting pretty quick, so I abandoned the idea.

File: scripts/docs.js [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+}
+
+function createMethod(file, block) {
+  return {
+    metadata: {
+      constructor: !!block.isConstructor,
+      name: detectName(block),
+      source: file.replace(/^\./, '') + '#L' + block.codeStart,
+      description: formatHtml(block.description.full),
+      // examples: getByType(block, 'example').map(createExample),
```
Comment (stephenplusplus) [inline scripts/docs.js]: Unneeded?

File: scripts/docs.js [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+      var docs = parseDocs(file, contents, 0);
+      var outputFile = path.join(
+        './docs/json/master',
+        file.replace('/lib', '').replace('.js', '.json')
+      );
+
+      fs.writeFile(outputFile, JSON.stringify(docs), callback);
+    });
+  });
```
Comment (stephenplusplus) [inline scripts/docs.js]: I might be reading this wrong, but should we add a callback here:

```js
async.each(files, function(file, callback) {/*...*/}, function(err) {
  if (err) {
    throw err;
  }
});
```

File: scripts/docs.sh [hunk 6]
```diff
@@ -1,73 +0,0 @@
-#!/bin/bash
```
Comment (stephenplusplus) [inline scripts/docs.sh]: Bye, weird file!

File: scripts/docs.js [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+ */
+
+'use strict';
+
+var globby = require('globby');
+var async = require('async');
+var fs = require('fs');
+var dox = require('dox');
+var path = require('path');
+var exec = require('propprop');
```
Comment (stephenplusplus) [inline scripts/docs.js]: We usually call this `prop`. `exec` is for `methmeth`. The difference is `prop` just returns the property, where `exec` will execute the method.

File: scripts/docs.js (hunk 4 above)
Comment (callmehiphop) [inline scripts/docs.js]: Ahh, yeah. I ended up backing out the JSDoc `<caption>` stuff. Thanks!

File: scripts/docs.js (hunk 7 above)
Comment (callmehiphop) [inline scripts/docs.js]: Gotcha, good call!

**Comment by quartzmo** [comment]
@callmehiphop Can you point me to where your build waits "for new doc app to be merged into gcloud-common"? I saw that got checked off but can't find the commit. I'm currently working on our gcloud-ruby json output using the docs app & assets that I manually copied from your gh-pages... But of course I need to automate.

**Comment by callmehiphop** [comment]
@quartzmo I was just waiting for the main app to get merged into gcloud-common, which it did yesterday via https://github.com/GoogleCloudPlatform/gcloud-common/pull/34. We have a hook that will generate JSON, etc. and didn't want to merge these changes in until all the other parts were good to go.

**Pull request merged**